CountryTweetMap/scripts/app.js

```diff
@@ -0,0 +1,173 @@
+    $scope.displayMap = function() {
+        var tweetCount = [];
+        $scope.tweetMaxCountries = [];
+        for (var key in $scope.tweetFreq) {
+            tweetCount.push($scope.tweetFreq[key].length);
+        }
+        $scope.tweetMax = Math.max.apply(null, tweetCount);
+        $scope.tweetMin = Math.min.apply(null, tweetCount);
+        for (var key in $scope.tweetFreq) {
+            if ($scope.tweetFreq[key].length === $scope.tweetMax) {
+                $scope.tweetMaxCountries.push($scope.tweetFreq[key][0].place_country);
+            }
+        }
+        $scope.tweetMaxCountries = $scope.tweetMaxCountries.join(",");
+        var rangeSize = $scope.tweetMax - $scope.tweetMin;
+        var rangeMid = ($scope.tweetMax + $scope.tweetMin) / 2;
+
+        var countriesHigh = [];
+        var countriesMedium = [];
+        var countriesLow = [];
+        for (var key in $scope.tweetFreq) {
+            var country = $scope.tweetFreq[key][0];
+            var size = $scope.tweetFreq[key].length;
+            if (size === $scope.tweetMax) {
+                var marker = L.marker(country.place_country_center, {icon: redIcon}).bindPopup(
+                    "<div class='header'>" + country.place_country +
+                    " (" + country.place_country_code + ")" + "</div><div class='tweet-number'>" +
+                    "Number of tweets : " + $scope.tweetFreq[key].length + "</div>"
+                );
+                countriesHigh.push(marker);
+                continue;
+            }
+            if (size > rangeMid) {
+                var marker = L.marker(country.place_country_center, {icon: blueIcon}).bindPopup(
+                    "<div class='header'>" + country.place_country +
+                    " (" + country.place_country_code + ")" + "</div><div class='tweet-number'>" +
+                    "Number of tweets : " + $scope.tweetFreq[key].length + "</div>"
+                );
+                countriesMedium.push(marker);
+                continue;
+            }
+            if (size <= rangeMid) {
+                var marker = L.marker(country.place_country_center, {icon: greenIcon}).bindPopup(
+                    "<div class='header'>" + country.place_country +
+                    " (" + country.place_country_code + ")" + "</div><div class='tweet-number'>" +
+                    "Number of tweets : " + $scope.tweetFreq[key].length + "</div>"
+                );
+                countriesLow.push(marker);
+                continue;
+            }
+        }
+        var countryHighGroup = L.layerGroup(countriesHigh);
+        var countryMediumGroup = L.layerGroup(countriesMedium);
+        var countryLowGroup = L.layerGroup(countriesLow);
+
+        var background = L.tileLayer(
+            'http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+            {maxZoom: 18});
+        $scope.map = L.map('map', {
+            center: [mapCenterLat, mapCenterLon],
+            zoom: 1,
```

Comment:
Change the value here, so that there is only 1 map, instead of 3. 3 maps looks weird.